Erictr/inkv2 (#195)
BonsaiAI/bonsaiai.github.io: source/includes/inkling2-guide/_concepts.html.md

* `active_tetrimino`: the tetrimino (piece) that is active.
* `tetrimino_location`: the location of the active tetrimino.
* `tetrimino_speed`: how fast the tetrimino is moving.
* `clear_base`: fit tetriminos into the base to clear pieces of it.

Any of these could be written into supporting concepts that would give information to the output concept.

### Example: Home Automation

Let's imagine that we want to save energy by turning off the lights in empty rooms. The output concept may be called `turn_off_lights_empty_rooms`. To support the output concept, we need to think about other concepts that may assist the AI. Here's a short list, but you might be able to think of other facts that you'd want to use:

* `light_state`: the on/off state of the lights in each room.
* `people_home`: The total number of people in the house.
* `person_location`: The location of each person/the room each person is in.

> * `done_with_it`: The amount of time a person has been out of a room. Or, possibly, the location of the person who left this room.

Reviewer: ```suggestion
* `time_empty`: The amount of time a room has been empty. 
```

* `definitely_empty`: don't shut off lights if someone is coming right back.
* `still_in_use`: (a variant of the one above) don't turn off the lights if an empty room might be reoccupied soon.

Any of these could be written into supporting concepts that would give information to the output concept.

## Writing Concept Statements

In Inkling, concepts are declared using the `concept` keyword followed by the concept's name, input list and output type. For readability, a descriptive name is recommended.

> Every concept must include a curriculum, which tells the AI engine how to train the concept. These will be covered below.

Reviewer: ```suggestion
Every learned concept must include a curriculum, which tells the AI engine how to train the concept. These will be covered below.
```
